### DISCUSSION TITLE
Trying to move from 'regular' docker-compose installation to Unraid - any ideas?

### DISCUSSION THREAD

**Discussion by Cantello** [immich-app/immich#12850] [Q&A] [answered]
Hi all!
I am in the process of moving my server installation to one unified server (before: NAS & separate docker server) and am stuck while moving Immich to the new server. I have backed up the database and restored it but apparently the paths are different on the two setups (`/usr/src/app/upload` before and `/photos` on Unraid). I tried editing these but still Immich does not find the photos.
Would it be a possibility to use a certain storage template that includes the album (which I would very much to preserve, spent quite a few hours organising the photos in albums) and then simply re-import the photos via CLI on the new installation?

**Comment by bo0tzz** [comment, marked as answer]
Immich doesn't yet cope very well with the internal path changing. You can mount your files to `/usr/src/app/upload` and also set `IMMICH_MEDIA_LOCATION` to that path.

**Comment by radh21301** [comment]
I would suggest you go the docker compose method for Unraid as well.  Give a lot more flexibility for these kinds of things. And honestly, much easier than the unraid app store templates for multi-container setups.

https://immich.app/docs/install/unraid/

**Comment by Cantello** [comment]
Just for the record, I did not manage to move Immich to the new system and the unraid community app really is different from the official docker one. I just used docker-compose on unraid and recreated the whole library. Sadly, the albums did not make it. I tried exporting them into folders but managed to eff that up as well. *shrug*

**Comment by munitoon** [comment]
I just had the same issue (but going from Unraid to docker-compose) and wrote this documentation for future reference. You should be able to use a similar approach to go from docker-compose to Unraid by replacing `/usr/src/app/upload/` with `/photos/`. The only catch may be in restoring the DB file to the Unraid immich-postgres container. It looks like you will need to remove both containers and delete the PostgreSQL_Immich data before reinstalling the PostgreSQL_Immich container, restore the db backup, then reinstall the immich container. Proceed with caution however, as I haven't tried it this way.

# Dump Existing DB

Follow the immich documentation to create a backup of the existing database:

[https://immich.app/docs/administration/backup-and-restore/](https://immich.app/docs/administration/backup-and-restore/ (preview))

Or, navigate to the backups folder and choose one of the automatic backups from there:

```
ls /mnt/user/photos_immich/backups/
```

# Edit the DB

The Unraid app stores photos in the directory `/photos` inside the docker container, whereas the docker-compose version store them in the directory `/usr/src/app/upload`. The database references all point to the `/photos` path and need to be updated to point to the `/usr/src/app/upload` path. 

There is probably a SQL command to accomplish this, but after looking through the manual and trying to understand how to do it properly, I found it easier to just open the database file in an editor and use the find/replace tool. 

Unzip and open the .sql file in VS Code (or other editor), enter `CTRL+f` and click the drop down arrow to show the "replace" field:

In my case, the database was too large to "replace all" in one go, so I highlighted a few large sections of the file and used the "Find in Selection" option (`ALT+l`). I'm not sure what the limit is, but I was able to do it in 3 or 4 selections.

![image](https://github.com/user-attachments/assets/e96a3680-3c48-455f-818f-9d714e6c0c57)

> [!WARNING]
I am using Authentik for OAuth, and I named my application "photos". The first time I attempted this, I accidentally broke my authentication redirect url because it contained "/photos/" and was replaced in this operation.
>  

Gzip the file and upload to a location accessible from the machine running the new immich instance.

# Restore the DB

Now, follow the immich "Manual Backup and Restore" documentation using the edited immich-db-backup.sql.gz file:

[https://immich.app/docs/administration/backup-and-restore/](https://immich.app/docs/administration/backup-and-restore/ (preview))
